File: optic/components/foo_bar/foo_bar.ts
import { Component, Input } from "@angular/core";
import * as pb from "optic/protos/ui_ts_proto_pb/protos/ui_pb";
import { ChannelService } from "../../../web/src/services/channel_service";

@Component({
  selector: "app-foo-bar",
  templateUrl: "foo_bar.html",
  standalone: true,
})
export class FooBarComponent {
  @Input() config!: pb.FooBarComponent;

  constructor(private readonly channelService: ChannelService) {}

  handleClick(event: any) {}
}

File: optic/components/foo_bar/foo_bar.py
import protos.ui_pb2 as pb
from optic.component_helpers import insert_component


def foo_bar(*, foo: str):
    """
    This function creates a button component with a label and an on_click event.

    Args:
        label (str): The text to be displayed on the button.
        on_click (Callable[..., Any]): The function to be called when the button is clicked.
    """
    insert_component(
        data=pb.ComponentData(
            foo_bar=pb.FooBarComponent(
                foo=foo,
            )
        )
    )
